backend/libraries/utils/src/text_validation.rs

```diff
@@ -0,0 +1,184 @@
+use types::{FieldTooLongResult, FieldTooShortResult, GroupSubtype};
+
+const MIN_USERNAME_LENGTH: u16 = 5;
+const MAX_USERNAME_LENGTH: u16 = 25;
+const MIN_GROUP_NAME_LENGTH: u32 = 3;
+const MAX_GROUP_NAME_LENGTH: u32 = 25;
+const MAX_GROUP_DESCRIPTION_LENGTH: u32 = 1024;
+const MAX_GROUP_RULES_LENGTH: u32 = 1024;
+
+pub enum UsernameValidationError {
+    TooLong(u16),
+    TooShort(u16),
+    Invalid,
+}
+
+pub fn validate_username(username: &str, ascii_only: bool) -> Result<(), UsernameValidationError> {
```

Comment:
We might easily want to have different min/max lengths for username/displayname. In fact I think the min for display name should be shorter - say 3 - and the max for username should bd reduced to say 18.

Reply:
This PR didn't change any functionality in that area, so we could decide exactly what we want to do and make that change as a separate PR?

Reply:
Display name validation is now updated.